app/javascript/components/UserProfileMenu/index.jsx

```diff
@@ -0,0 +1,144 @@
+
+import UpdateUserOfficeMutation from '/mutations/UpdateUserOfficeMutation.gql'
+import { UserContext, FilterContext } from '/context'
+import { DefaultOfficeMenu, LanguageMenu, GeneralSettingsMenu } from './Menus'
+
+const UserDetails = styled.div`
+  margin-left: ${({ theme }) => theme.styles.spacing.sm};
+`
+
+const UserProfileContainer = styled.button`
+  display: flex;
+  border: none;
+  text-align: left;
+`
+
+const UserName = styled(MD)`
+  font-weight: bold;
+`
+
+const UserProfileMenu = ({ offices, location, router, togglePopover }) => {
+  const { currentUser, setOffice } = useContext(UserContext)
+  const { setOfficeValue } = useContext(FilterContext)
+  const [ updateDefaultOffice ] = useMutation(UpdateUserOfficeMutation)
+
+  if (R.isNil(currentUser) || R.isEmpty(currentUser)) return null
+
+  const handleOfficeSelect = office =>
+    updateDefaultOffice({ variables: { userId: currentUser.id, officeId: office.id } })
+      .then(response => setOffice(R.path(['data', 'updateUserOffice', 'office'], response)))
+      .then(_ => setOfficeValue(office.id))
+      .then(_ => togglePopover('user'))
+
+  const { i18n, t } = useTranslation()
+  const languages = [
+    { label: 'English', value: 'en' },
+    // Enable when Japanense is supported
+    // { label: '日本語', value: 'ja' }
+  ]
+
+  const [isOpen, setIsOpen] = useState(false)
+  const [tempSelectedItem, setTempSelectedItem] = useState()
+  const [selectedItem, setSelectedItem] = useState({ office: currentUser.office, language: languages[0] })
+
+  const handleStateChange = (changes, stateAndHelpers) => {
+    if (R.hasPath(['isOpen'])(changes)) {
+      console.log(changes.selectedItem)
+      setIsOpen(
+        changes.selectedItem === 'default-office' ||
+        changes.selectedItem === 'language-settings' ||
+        changes.selectedItem === 'general-settings' ||
+        changes.isOpen
+      )
+    }
+
+    if (R.hasPath(['selectedItem'])(changes)) {
+      const itemSelected = R.prop('selectedItem')(changes)
+      if (itemSelected === 'general-settings') {
+        switch (tempSelectedItem) {
+          case 'default-office':
+            stateAndHelpers.setHighlightedIndex(0)
+            break
+          case 'language-settings':
+            stateAndHelpers.setHighlightedIndex(1)
+        }
+      }
+      setTempSelectedItem(itemSelected)
+    }
+  }
+
+  const handleSelect = selectedItem => {
+    if (tempSelectedItem === 'default-office') {
+      if (R.hasPath(['office'])(selectedItem)) {
+        handleOfficeSelect(selectedItem.office)
+        setSelectedItem(selectedItem)
+      }
+    } else if (tempSelectedItem === 'language-settings') {
+      if (R.hasPath(['language'])(selectedItem)) {
+        i18n.changeLanguage(selectedItem.language.value, () => {
+          // TODO: Handle callback (error/success)
+        })
+        setSelectedItem(selectedItem)
+      }
+    } else if (selectedItem === 'admin') {
+      router.push('/portal/admin')
```

Comment:
if the menu items themselves are router links, do we need to do this here?

Reply:
it's for accessibility - since when a user selects this item (via tabbing etc), it will trigger the select handler instead of the Link

I'll add a comment for this